src/components/Footer/JoinUs/JoinUs.js

```diff
@@ -0,0 +1,68 @@
+import React, { Component } from 'react';
+import './JoinUs.scss';
+import '../Footer.scss';
+
+class JoinUs extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: '',
+      formError: {
+        email: '',
+      },
+      emailValid: false,
+      formValid: false,
+    };
+  }
+
+  handleInput(e) {
+    e.preventDefault();
+    const name = e.target.name;
+    const value = e.target.value;
+    this.setState({ [name]: value }, () => {
+      this.validateField(value);
+    });
+  }
+
+  validateField(value) {
+    const emailRegEx = /^(\D)+(\w)*((\.(\w)+)?)+@(\D)+(\w)*((\.(\D)+(\w)*)+)?(\.)[a-z]{2,}$/;
+    const fieldValidationError = this.state.formError;
+    let emailValid = this.state.emailValid;
+
+    emailValid = value.match(emailRegEx);
+    fieldValidationError.email = emailValid ? '' : ' is invalid';
+
+    this.setState({ formError: fieldValidationError,
+      emailValid: emailValid,
+    }, this.validateForm);
```

Comment:
I guess this callback is unnecessary here - you can just set one more prop to state, it will cause just 1 rerender instead of 2, because only thing that validateForm do  is setting state
What do you think about it?